src/cortxfs/cortxfs_ops.c

```diff
@@ -550,15 +552,32 @@ int cfs_destroy_orphaned_file(struct cfs_fs *cfs_fs,
 	kvs_end_transaction(kvstor, &index);
 
 out:
-	kvnode_fini(&node);
-	if (stat) {
-		kvs_free(kvstor, stat);
-	}
-
 	if (rc != 0) {
 		kvs_discard_transaction(kvstor, &index);
 	}
 
+	log_trace("inode=%llu rc=%d", *ino, rc);
+	return rc;
+}
+
+int cfs_destroy_orphaned_file(struct cfs_fs *cfs_fs,
+                              const cfs_ino_t *ino)
+{
+	int rc;
+	struct cfs_fh *fh = NULL;
+
+	/* TODO:Temp_FH_op - to be removed
+	 * Should get rid of creating and destroying FH operation in this
+	 * API when caller pass the valid FH instead of inode number
+	 */
+	RC_WRAP_LABEL(rc, out, cfs_fh_from_ino, cfs_fs, ino, &fh);
+	RC_WRAP_LABEL(rc, out, cfs_destroy_orphaned_file2, fh);
+
+out:
+	if (fh != NULL) {
+		cfs_fh_destroy(fh);
+	}
+	log_trace("inode=%llu rc=%d", *ino, rc);
```

Comment:
Do we need to check for NULL before accessing *ino?

Reply:
`cfs_fh_from_fh` is check here and caller of this APIs are also making sure hence no check. Also here using `ino` we are deriving `fh` and that is also validated in `cfs_fh_from_inode` before returning here. So I guess no validation is required.